File: TwoSum/hash-table/twoSum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_map = {}
        for i in range(len(nums)):
            current = nums[i]
            x = target - current
            if x in hash_map:
                return [hash_map[x], i]
            hash_map[nums[i]] = i
        

File: TwoSum/hash-table/twoSum.java
import java.util.Map;
import java.util.HashMap;

public class twoSum {
    public int[] twosum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int current = nums[i];
            int x = target - current;
            if (map.containsKey(x)) {
                return new int[] {
                    map.get(x), i
                };
            }
            map.put(current, i);
        }
        return null;
    }
}